src/server/req_runjob.c

```diff
@@ -613,6 +613,18 @@ req_runjob2(struct batch_request *preq, job *pjob)
 	/* Check if prov is required, if so, reply_ack and let prov finish */
 	/* else follow normal flow */
 	prov_rc = check_and_provision_job(preq, pjob, &need_prov);
+
+	/* in case of subjob, it was never saved to the database so far.
+	 * Save it now, before a possiblity to return from the routine
+	 */
+	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
+		if (job_save(pjob, SAVEJOB_NEW)) {
```

Comment:
@Bhagat-Rajput  Will it be better if we avoid `job_save(subjob)` when `check_and_provision_job()` fails ? If you agree you can move the `if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {...}` block to location after `if (prov_rc) {...}` block below

Reply:
@Bhagat-Rajput and @Shrini-h - wouldn't we still need the subjob to be saved to db, even if we could not start provisioning for the subjob?

Reply:
@prakashcv13  you are right, I mistook the free_nodes() as job_purge(), hence I suggested it.
@Bhagat-Rajput  please revert this change and verify once with provision fail test when Job Array is submitted